provider/aws/config.go

```diff
@@ -80,46 +73,47 @@ func CreateSessions(cfg *externaldns.Config) map[string]*session.Session {
 			if err != nil {
 				logrus.Fatal(err)
 			}
-			result[profile] = session
+			result[profile] = cfg
 		}
 	}
 	return result
 }
 
-func newSession(awsConfig AWSSessionConfig) (*session.Session, error) {
-	config := aws.NewConfig().WithMaxRetries(awsConfig.APIRetries)
-
-	config.WithHTTPClient(
-		instrumented_http.NewClient(config.HTTPClient, &instrumented_http.Callbacks{
+func newV2Config(awsConfig AWSSessionConfig) (awsv2.Config, error) {
```

Comment:
why not return `(*awsv2.Config, error)`?

Reply:
aws-sdk-go-v2 service clients expect an `awsv2.Config` as input, https://github.com/kubernetes-sigs/external-dns/pull/4640/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R243 and https://github.com/kubernetes-sigs/external-dns/pull/4640/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R414 

There's nothing stopping us from returning `*awsv2.Config`, but we would immediately dereference it each time it needs to be used

ref: https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#newfromconfig